src/metalnx-web/src/main/java/com/emc/metalnx/controller/TemplateController.java

```diff
@@ -302,10 +312,27 @@ public String modifyTemplate(final Model model, @ModelAttribute final MetadataTe
 			}
 			DataGridUser loggedUser = loggedUserUtils.getLoggedDataGridUser();
 
-			// check if user is the template owner or admin
-			if (!(template.getOwner().equalsIgnoreCase(loggedUser.getUsername()) || loggedUser.isAdmin())) {
-				throw new Exception("Cannot modify a template beloging to another user");
+			// throw error if user is not the template owner, rodsadmin or groupadmin
+ 			if (!(template.getOwner().equalsIgnoreCase(loggedUser.getUsername())) && loggedUser.isAdmin() == false && loggedUser.isGroupAdmin() == false) {
+				throw new Exception("Cannot modify a template belonging to another user");
 			}
+			// throw error if logged user is not the rodsadmin of the template owner's group	
+			if(loggedUser.isGroupAdmin()){
+			boolean groupAdminEditPermission = false;
+			String[] templateOwnerGroupList = userService.getGroupIdsForUser(template.getOwner(), irodsServices.getCurrentUserZone());
+			Set<String> groupAdminGroupList = new HashSet<>(Arrays.asList(userService.getGroupIdsForUser(loggedUser.getUsername(), irodsServices.getCurrentUserZone())));
+			logger.info("groupList length is: " + templateOwnerGroupList.length);
+			for(int i = 0; i< templateOwnerGroupList.length && !groupAdminEditPermission; i++){
+				logger.info("current user is: " + templateOwnerGroupList[i]);
+				if(groupAdminGroupList.contains(templateOwnerGroupList[i])) groupAdminEditPermission = true;
+			}
+			if(!groupAdminEditPermission) throw new Exception("Cannot modify a template belonging to a user of other group");
+			
```

Comment:
please indent 321-330

please break 327 and 329 into three lines each (using brackets to make the if blocks safer)